### PR TITLE
[Implements #9] Default to using ffmpeg for audio and screenshots, fallback to mpv if ffmpeg fails

### LINKED ISSUE #9: [FEATURE] Use ffmpeg instead of mpv to generate card picture/audio
Some mpv builds don't contain the needed codecs to generate png/jpg images.

### PR DESCRIPTION
**Pull request by christofferaakre**
Not sure if you guys accept pull requests, but I had a go at implementing #9. Sorry for the messy diff. Let me know if there is anything I should change before you can merge this!
I belive this fixes #9

### CODE REVIEW

**Comment by RicBent** [comment]
Hey there! Thanks for the contribution.

Actually this was already somewhat implemented in https://github.com/migaku-official/migaku-mpv/tree/ffmpeg-export

The main reason this was not yet merged is that migaku mpv actually allows creating cards from web resources like YouTube and Viki. Switching to ffmpeg breaks that use case. What might work however is first attempting to export the media with ffmpeg and then use mpv as a fallback on failure.

**Comment by christofferaakre** [comment]
> Hey there! Thanks for the contribution.
> 
> Actually this was already somewhat implemented in https://github.com/migaku-official/migaku-mpv/tree/ffmpeg-export
> 
> The main reason this was not yet merged is that migaku mpv actually allows creating cards from web resources like YouTube and Viki. Switching to ffmpeg breaks that use case. What might work however is first attempting to export the media with ffmpeg and then use mpv as a fallback on failure.

I see. The code I wrote does actually fallback to mpv if ffmpeg fails

**Comment by RicBent** [comment]
Oh I'm actually blind. Sorry for that.

Will check it out and merge it asap.

File: utils/ankiexport.py [hunk 1]
```diff
@@ -80,20 +88,92 @@ def export_card(self, media_file, audio_track, text_primary, text_secondary, tim
...
+            error, proc = self.mpv_audio(media_file, audio_track, start, end, out_path)
+
+        if error:
+            error = Errors.AUDIO_ERROR
+
+        return error, proc
+
+    def ffmpeg_screenshot(self, media_file, start, end, out_path):
+        args = [
+                'ffmpeg',
```
Comment (RicBent) [inline utils/ankiexport.py:142]: This should use the ffmpeg executable from https://github.com/migaku-official/migaku-mpv/blob/063fe424754af72350d2ad6935c5bd57d35a1d3d/migaku_mpv.py#L783

File: utils/ankiexport.py [hunk 2]
```diff
@@ -80,20 +88,92 @@ def export_card(self, media_file, audio_track, text_primary, text_secondary, tim
             return -2
 
         return 0
-        
 
-
-    def make_audio(self, media_file, audio_track, start, end, out_path):
+    def ffmpeg_audio(self, media_file, audio_track, start, end, out_path):
+        args = [
+                'ffmpeg',
```
Comment (RicBent) [inline utils/ankiexport.py:94]: This should use the ffmpeg executable from https://github.com/migaku-official/migaku-mpv/blob/063fe424754af72350d2ad6935c5bd57d35a1d3d/migaku_mpv.py#L783

**Review by RicBent — CHANGES_REQUESTED** [review]
This looks fine!

Just this thing needs to be fixed before I'll merge it. Especially Windows users wouldn't have `ffmpeg` in `PATH`.

**Comment by christofferaakre** [comment]
@RicBent just doing `from migaku_mpv import find_executable` in `ankiexport.py` doesn't work since they are not in the same directory, and I don't really want to mess with the directory strucutre. I see there is an `__init__.py` file in `utils`, but I am not quite sure what the package structure is supposed to be like.

**Comment by RicBent** [comment]
The best solution probably would be handling it the same way as `mpv_executable` as that file was meant as something independent that could be reused in other projects later down the line. References for how `mpv_executable` is set:

https://github.com/migaku-official/migaku-mpv/blob/063fe424754af72350d2ad6935c5bd57d35a1d3d/migaku_mpv.py#L490
https://github.com/migaku-official/migaku-mpv/blob/063fe424754af72350d2ad6935c5bd57d35a1d3d/utils/ankiexport.py#L12

**Comment by RicBent** [comment]
Thanks a lot! Merging~

**Comment by christofferaakre** [comment]
Great! Thanks for being patient with me

**Pull request merged**